### PR TITLE
Add PySpark Example For XGBoost Multi-Class Classification on MNIST

### PR DESCRIPTION
**Pull request by robperc**

### CODE REVIEW

File: sagemaker-pyspark-sdk/README.rst [hunk 1]
```diff
@@ -156,6 +156,71 @@ parameter.
 
 You can view the `PySpark API Documentation for SageMaker Spark here <http://sagemaker-pyspark.readthedocs.io/en/latest/>`_
 
+
+Training and Hosting an XGBoost model using SageMaker PySpark
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+A XGBoostSageMakerEstimator runs a training job using the Amazon SageMaker XGBoost algorithm upon
```
Comment (iquintero) [inline sagemaker-pyspark-sdk/README.rst]: An XGBoost....

**Comment by robperc** [comment]
I added Jupyter notebooks for this example showing how to build the XGBoost model then invoke it from Spark and boto3. Would you prefer to remove the example from the pyspark SDK README and push it to the notebooks instead? 

Sorry for the slop on this I should have done this in two separate PRs in multiple branches but it slipped my mind.

File: sagemaker-pyspark-sdk/README.rst [hunk 2]
```diff
@@ -156,6 +156,71 @@ parameter.
...
+    xgboost_estimator.setNumClasses(10)
+
+    xgboost_model = xgboost_estimator.fit(training_data)
+
+    transformed_data = xgboost_model.transform(test_data)
+    transformed_data.show()
+
+The SageMakerEstimator expects an input DataFrame with a column named "features" that holds a
+Spark ML  Vector. The estimator also serializes a "label" column of Doubles if present. Other
+columns are ignored. The dimension of this input vector should be equal to the feature dimension
```
Comment (andremoeller) [inline sagemaker-pyspark-sdk/README.rst]: > The dimension of this input vector should be equal to the feature dimension given as a hyperparameter

Could you remove this sentence? XGBoost doesn't have a feature dimension hyperparameter.

File: sagemaker-pyspark-sdk/README.rst [hunk 3]
```diff
@@ -156,6 +156,71 @@ parameter.
...
+The Amazon SageMaker XGBoost algorithm accepts many parameters. Objective (the learning objective of your model, in this case multi-class classification) and NumRounds (the number of rounds to perform tree boosting on) are required. For multi-class classification NumClasses (the number of classes to classify the data into) is required as well.
+
+You can set other hyperparameters, for details on them, run:
+
+.. code-block:: python
+
+    xgboost_estimator.explainParams()
+
+After training is complete, an Amazon SageMaker Endpoint is created to host the model and serve
+predictions. Upon invocation of transform(), the SageMakerModel predicts against their hosted
```
Comment (andremoeller) [inline sagemaker-pyspark-sdk/README.rst]: > against their

-> "against the"

File: sagemaker-pyspark-sdk/README.rst [hunk 4]
```diff
@@ -156,6 +156,71 @@ parameter.
...
+model. Like the SageMakerEstimator, the SageMakerModel expects an input DataFrame with a column
+named "features" that holds a Spark ML Vector equal in dimension to the value of the FeatureDim
```
Comment (andremoeller) [inline sagemaker-pyspark-sdk/README.rst]: Similarly, could you remove the "FeatureDim" part?

**Comment by robperc** [comment]
Good catch! Missed those when I copied over the skeleton text from the previous example :P

**Comment by robperc** [comment]
[This commit](5d64b29) updates the url for the data for the test invocation to point to where it will be in the aws/sagemaker-spark repo after this PR is merged. Just want to point this out because the notebook won't work until the PR is merged since the data file won't be at that url until then.

**Pull request merged**